src/Pos/Explorer/Web/Server.hs

```diff
@@ -126,6 +142,24 @@ explorerHandlers _sendActions =
     defaultLimit limit = (fromIntegral $ fromMaybe 100 limit)
     defaultSkip  skip  = (fromIntegral $ fromMaybe 0 skip)
 
+-- | Get the total number of blocks/slots currently available.
+-- Total number of main blocks   = difficulty of the topmost (tip) header.
+-- Total number of anchor blocks = current epoch + 1
+getBlocksTotalNumber
+    :: (MonadDB m)
+    => m Int
+getBlocksTotalNumber = do
+    -- Get the tip block.
+    tipBlock <- DB.getTipBlock @SscGodTossing
+    -- -1 is for the genesis block I guess?
+    pure $ checkMaxBlocks $ maxBlocks tipBlock - 1
```

Comment:
You are right. The expression is simpler this way. 
I find `pure $ max 0 (maxBlocks tipBlock - 1)` more readable.